processing/src/main/java/org/apache/carbondata/processing/datatypes/GenericDataType.java

```diff
@@ -32,7 +33,7 @@
  * Generic DataType interface which will be used while data loading for complex types like Array &
  * Struct
  */
-public interface GenericDataType<T> {
+public interface GenericDataType<T> extends Serializable {
```

Comment:
There are some logger in subclass, make them transient

Reply:
ok done

Reply:
Primitive type logger was not user, removed the field itself